fern/pages/src/providers/py-custom.mdx

```diff
@@ -1,6 +1,540 @@
+
+# Create an agentic provider
+make create-provider name=myagent agentic=true
+```
+
+This will create a new provider in `python/providers/<provider-name>/` with:
+- Complete package structure with `pyproject.toml` and `setup.py`
+- Provider implementation template
+- Demo script
+- README with usage examples
+- Type annotations and proper inheritance
+
+<Tip>
+The scaffolding script creates a fully functional provider template. You just need to implement the tool transformation logic specific to your platform.
+</Tip>
+
+### Generated structure
+
+The create-provider script generates the following structure:
+
+```
+python/providers/<provider-name>/
+├── README.md                    # Documentation and usage examples
+├── pyproject.toml              # Poetry configuration
+├── setup.py                    # Setup script for pip compatibility
+├── <provider-name>_demo.py     # Demo script showing usage
+└── composio_<provider-name>/   # Package directory
+    ├── __init__.py             # Package initialization
+    ├── provider.py             # Provider implementation
+    └── py.typed                # PEP 561 type marker
+```
+
+After generation, you can:
+1. Navigate to the provider directory: `cd python/providers/<provider-name>`
+2. Install in development mode: `uv pip install -e .`
+3. Implement your provider logic in `composio_<provider-name>/provider.py`
+4. Test with the demo script: `python <provider-name>_demo.py`
+
+## Creating a non-agentic provider
+
+Non-agentic providers inherit from the `NonAgenticProvider` abstract class:
+
+```python
+from typing import List, Optional, Sequence, TypeAlias
+from composio.core.provider import NonAgenticProvider
+from composio.types import Tool, Modifiers, ToolExecutionResponse
+
+# Define your tool format
+class MyAITool:
+    def __init__(self, name: str, description: str, parameters: dict):
+        self.name = name
+        self.description = description
+        self.parameters = parameters
+
+# Define your tool collection format
+MyAIToolCollection: TypeAlias = List[MyAITool]
+
+# Create your provider
+class MyAIProvider(NonAgenticProvider[MyAITool, MyAIToolCollection], name="my-ai-platform"):
+    """Custom provider for My AI Platform"""
+    
+    def wrap_tool(self, tool: Tool) -> MyAITool:
+        """Transform a single tool to platform format"""
+        return MyAITool(
+            name=tool.slug,
+            description=tool.description or "",
+            parameters={
+                "type": "object",
+                "properties": tool.input_parameters.get("properties", {}),
+                "required": tool.input_parameters.get("required", [])
+            }
+        )
+    
+    def wrap_tools(self, tools: Sequence[Tool]) -> MyAIToolCollection:
+        """Transform a collection of tools"""
+        return [self.wrap_tool(tool) for tool in tools]
+    
+    # Optional: Custom helper methods for your AI platform
+    def execute_my_ai_tool_call(
```

Comment:
add an example of how this method is supposed to accessed (ignore if already added)